### PR TITLE
clipping: store all GUI settings

### PR DESCRIPTION
**Pull request by edgomez**
- when using guides, store all required fields so from photo to photo
  the user has consistent GUI
- refactored code a bit in order to show correct parts of the UI when
  required

### CODE REVIEW

**Comment by edgomez** [comment]
could be considered for stable as well. This fixes an annoying UI design and should have no compatibility impact

**Comment by edgomez** [comment]
If no objection, will be merged tomorrow.

File: src/iop/clipping.c [hunk 1]
```diff
@@ -47,14 +47,13 @@ DT_MODULE_INTROSPECTION(5, dt_iop_clipping_params_t)
...
-#define NUM_GUIDES 8
-
 /** flip H/V, rotate an image, then clip the buffer. */
 typedef enum dt_iop_clipping_flags_t
 {
-  FLAG_FLIP_HORIZONTAL = 1,
-  FLAG_FLIP_VERTICAL = 2
+  FLAG_FLIP_NONE = 0,
+  FLAG_FLIP_HORIZONTAL = 1<<0,
+  FLAG_FLIP_VERTICAL = 2<<1,
```
Comment (houz) [inline src/iop/clipping.c]: Don't you mean `1<<1`?

Comment (edgomez) [inline src/iop/clipping.c]: Indeed thanks for catching this. I'll fix it this evening
Édouard Gomez
On 31 May 2015 16:04, houz <notifications@github.com> wrote:In src/iop/clipping.c:
>  /** flip H/V, rotate an image, then clip the buffer. */
>  typedef enum dt_iop_clipping_flags_t
>  {
> -  FLAG_FLIP_HORIZONTAL = 1,
> -  FLAG_FLIP_VERTICAL = 2
> +  FLAG_FLIP_NONE = 0,
> +  FLAG_FLIP_HORIZONTAL = 1<<0,
> +  FLAG_FLIP_VERTICAL = 2<<1,

Don't you mean 1<<1?

**Comment by edgomez** [comment]
@houz fixed !

thx. any other comment ?

**Pull request merged**